src/Compiler/Compiler.h

```diff
@@ -123,6 +123,8 @@ class Compiler {
   void SetIPGenerator(IPGenerator* generator) { m_IPGenerator = generator; }
   IPGenerator* GetIPGenerator() { return m_IPGenerator; }
   bool BuildLiteXIPCatalog(std::filesystem::path litexPath);
+  bool HasIPInstances();
+  bool HasIPDefinitions();
```

Comment:
why not const?

Reply:
Didn't think of it. I tried switching those two to const functions, but they both call GetIPGenerator() which causes a build error once those functions are made const and i don't think we want to make GetIPGenerator const.